ingress controller and dns mgt (#49752)
openshift/openshift-docs: networking/ingress-controller-dnsmgt.adoc

:_content-type: ASSEMBLY
[id="ingress-controller-dnsmgt"]
= Configuring an Ingress Controller for disabling DNS Management
include::_attributes/common-attributes.adoc[]
:context: ingress-controller-dnsmgt

toc::[]

As a cluster administrator, when you create an Ingress Controller, the Operator manages the DNS records by default. This has some limitations when the required DNS zone is different from the cluster DNS zone or when DNS zone is hosted outside the cloud provider. In such scenarios, the Operator reports a degraded state after attempting to provision the required resources in the cloud provider because the specified DNS can not be created.

Day 2 operations add support to the Operator to have the option to manage DNS record lifecycle on the cloud provider. As a cluster administrator, you can create or modify an Ingress Controller with `dnsManagementPolicy` to specify whether the Operator needs to manage the DNS record lifecycle or not.
You need this in instances where the required DNS zone is different from the cluster DNS zone and may not be hosted in the specific cloud provider.

[NOTE]
====

> This feature is supported on default or custom Ingress Controllers. As a clsuter administrator, you have the discretion to create or modify the default Ingress Controller.

Author: Could you please elaborate on the difference between non-default and custom? my thought is that when it is non-default, it means we are customizing it hence the word custom , unless i am not understanding.

> This feature is supported on default or custom Ingress Controllers. As a clsuter administrator, you have the discretion to create or modify the default Ingress Controller.

Reviewer: Non-default and custom mean the same, just that `non-default` is more explicit in what it conveys since this feature is NOT supported for the default ingress controller. 

If it makes it clear we can mention this the other way around, 
```
This feature is not supported on default Ingress Controllers 
...
```

:_content-type: ASSEMBLY
[id="ingress-controller-dnsmgt"]

> = Configuring an Ingress Controller for disabling DNS Management

Reviewer 2: see above

Author: changed to manual

include::_attributes/common-attributes.adoc[]
:context: ingress-controller-dnsmgt

toc::[]

> As a cluster administrator, when you create an Ingress Controller, the Operator manages the DNS records by default. This has some limitations when the required DNS zone is different from the cluster DNS zone or when the DNS zone is hosted outside the cloud provider. In such scenarios, the Operator reports a degraded state after attempting to provision the required resources in the cloud provider because the wildcard DNS record can not be created on the cloud provider.

Reviewer 2: ```suggestion
As a cluster administrator, when you create an Ingress Controller, the Operator manages the DNS records automatically. This has some limitations when the required DNS zone is different from the cluster DNS zone or when the DNS zone is hosted outside the cloud provider. 
```

> As a cluster administrator, you can create or modify an Ingress Controller with `dnsManagementPolicy` to specify if the Operator needs to manage the DNS record lifecycle.

Reviewer 2: ```suggestion
As a cluster administrator, you can configure an Ingress Controller to stop automatic DNS management and assume manual DNS management duties.   Set `dnsManagementPolicy` to specify when it should be automatically or manually managed.
```

== `Managed` DNS management policy

> The `Managed` DNS management policy for Ingress Controllers ensures that the lifecycle of the wildcard DNS record on the cloud provider is completely managed by the Operator.

Reviewer 2: ```suggestion
The `Managed` DNS management policy for Ingress Controllers ensures that the lifecycle of the wildcard DNS record on the cloud provider is automatically managed by the Operator.
```

> During transitioning an Ingress Controller from `Unmanaged` to `Managed` DNS management policy, the Operator attempts to create the DNS record on the cloud provider if it does not exist or updates the DNS record if it already exists.

Reviewer 2: Move tis to the `Unmanaged` section?

== `Unmanaged` DNS management policy

> The `Unmanaged` DNS management policy for Ingress Controllers ensures that the lifecycle of the wildcard DNS record on the cloud provider is not managed by the Operator, instead it is the responsibility of the cluster administrator.

Reviewer 2: ```suggestion
The `Unmanaged` DNS management policy for Ingress Controllers ensures that the lifecycle of the wildcard DNS record on the cloud provider is not automatically managed, instead it becomes the responsibility of the cluster administrator.
```

> During transitioning an Ingress Controller from `Managed` to `Unmanaged` DNS management policy, the Operator does not clean up the previous wildcard DNS record provisioned on the cloud.

Reviewer 2: Move this to the `Managed` section?

> Note that on the AWS cloud platform, if the domain on the Ingress Controller does not match with `dnsConfig.Spec.BaseDomain` then the DNS management policy is automatically set to `Unmanaged`.

Reviewer 2: Make this a special Note.

:_content-type: ASSEMBLY
[id="ingress-controller-dnsmgt"]
= Configuring an Ingress Controller for disabling DNS Management
include::_attributes/common-attributes.adoc[]
:context: ingress-controller-dnsmgt

toc::[]

As a cluster administrator, when you create an Ingress Controller, the Operator manages the DNS records by default. This has some limitations when the required DNS zone is different from the cluster DNS zone or when DNS zone is hosted outside the cloud provider. In such scenarios, the Operator reports a degraded state after attempting to provision the required resources in the cloud provider because the wildcard DNS record can not be created on the cloud provider.

> Day 2 operations add support to the Operator to have the option to manage DNS record lifecycle on the cloud provider. As a cluster administrator, you can create or modify an Ingress Controller with `dnsManagementPolicy` to specify whether the Operator needs to manage the DNS record lifecycle or not.

Reviewer: ```suggestion
As a cluster administrator, you can create or modify an Ingress Controller with `dnsManagementPolicy` to specify whether the Operator needs to manage the DNS record lifecycle or not.
```

Reviewer: nit: This seems redundant in my opinion.

Author: Removed

== `Managed` DNS management policy
The `Managed` DNS management policy for Ingress Controllers ensures that the lifecycle of the wildcard DNS record on the cloud provider is completely managed by the Operator.
During transitioning an Ingress Controller from `Unmanaged` to `Managed` DNS management policy, the Operator attempts to create the DNS record on the cloud provider if it does not exist or updates the DNS record if it already exists.

== `Unmanaged` DNS management policy
The `Unmanaged` DNS management policy for Ingress Controllers ensures that the lifecycle of the wildcard DNS record on the cloud provider is not managed by the Operator, instead it is the responsibility of the cluster administrator.
During transitioning an Ingress Controller from `Managed` to `Unmanaged` DNS management policy, the Operator does not clean up the previous wildcard DNS record provisioned on the cloud.

Note that on the AWS cloud platform, if the domain on the Ingress Controller does not match with `dnsConfig.Spec.BaseDomain` then the DNS management policy is automatically set to `Unmanaged`.

[NOTE]
====

> This feature is not supported on default Ingress Controllers. As a cluster administrator, you have the discretion to create or modify the default Ingress Controller.

Reviewer 3: Does "This feature" refer to managed, unmanaged, or both? Probably better to be more specific.

Author: Re-phrasing

Author: The feature is being able to configure the DNS management policy.  and the two values are managed and unmanaged.  @thejasn can correct me if I'm wrong.

Reviewer: Yes, it refers to this user configurable field. Maybe we re-word it as,

> Updating `dnsManagementPolicy` is not supported on the default Ingress Controllers. As a cluster administrator, you have the discretion to create or modify the default Ingress Controller.

Reviewer 3: Thanks, @thejasn and @Amrita42. That would work for me!!

> ====

Reviewer: We can remove this note, we do support patching default ingress controllers as well.

:_content-type: ASSEMBLY
[id="ingress-controller-dnsmgt"]
= Configuring an Ingress Controller for manual DNS Management
include::_attributes/common-attributes.adoc[]
:context: ingress-controller-dnsmgt

toc::[]

As a cluster administrator, when you create an Ingress Controller, the Operator manages the DNS records automatically. This has some limitations when the required DNS zone is different from the cluster DNS zone or when the DNS zone is hosted outside the cloud provider.

As a cluster administrator, you can configure an Ingress Controller to stop automatic DNS management and start manual DNS management. Set `dnsManagementPolicy` to specify when it should be automatically or manually managed.

== `Managed` DNS management policy
The `Managed` DNS management policy for Ingress Controllers ensures that the lifecycle of the wildcard DNS record on the cloud provider is automatically managed by the Operator.

> During transitioning an Ingress Controller from `Managed` to `Unmanaged` DNS management policy, the Operator does not clean up the previous wildcard DNS record provisioned on the cloud.
> During transitioning an Ingress Controller from `Unmanaged` to `Managed` DNS management policy, the Operator attempts to create the DNS record on the cloud provider if it does not exist or updates the DNS record if it already exists.

Reviewer 2: Both of the lines "During transitioning..." (or a summary of the content) should be moved to where they will be most relevant.  Maybe at the first paragraph, the overview.
```suggestion
When you change an Ingress Controller from `Managed` to `Unmanaged` DNS management policy, the Operator does not clean up the previous wildcard DNS record provisioned on the cloud.
When you change an Ingress Controller from `Unmanaged` to `Managed` DNS management policy, the Operator attempts to create the DNS record on the cloud provider if it does not exist or updates the DNS record if it already exists.
```

== `Unmanaged` DNS management policy
The `Unmanaged` DNS management policy for Ingress Controllers ensures that the lifecycle of the wildcard DNS record on the cloud provider is not automatically managed, instead it becomes the responsibility of the cluster administrator.

[NOTE]
====
On the AWS cloud platform, if the domain on the Ingress Controller does not match with `dnsConfig.Spec.BaseDomain` then the DNS management policy is automatically set to `Unmanaged`.
====

[IMPORTANT]
====
When you set `dnsManagementPolicy` to `unmanaged`, you have to manually manage the lifecycle of the wildcard DNS record on the cloud provider. 

> ====

Reviewer 2: Can we put this `IMPORTANT` in a very noticeable place, at the top before they read (or skip) any other paragraphs?

Author: I have placed it in the overview
